src/sonic-config-engine/sonic-cfggen

```diff
@@ -229,6 +269,10 @@ def main():
     group.add_argument("--preset", help="generate sample configuration from a preset template", choices=get_available_config())
     group = parser.add_mutually_exclusive_group()
     group.add_argument("-K", "--key", help="Lookup for a specific key")
+    group.add_argument("-b", "--batch", help="Render and dump template data into a file (batch mode)", action="append", default=[],
+                       type=lambda opt_value: tuple(opt_value.split(',')))
+    group.add_argument("-J", "--process-json", help="render json file from a template and update internal switch config data",
```

Comment:
>-J [](start = 24, length = 2)

Could you also provide an example with the intention? It is not clear from the help string.

Reply:
Here is an example from the test I did:

```
/usr/local/bin/sonic-cfggen -d -J '/usr/share/sonic/device/x86_64-dell_s6000_s1220-r0/Force10-S6000/buffers.json.j2' -y '/etc/sonic/sonic_version.yml' -J '/usr/share/sonic/device/x86_64-dell_s6000_s1220-r0/Force10-S6000/qos.json.j2' --write-to-db
````
The above example would have been done in 4 calls into cfggen.

Reply:
Thanks for the sample!
I propose a command line:
```
/usr/local/bin/sonic-cfggen --write-to-db -d -y /etc/sonic/sonic_version.yml -t /usr/share/sonic/device/x86_64-dell_s6000_s1220-r0/Force10-S6000/buffers.json.j2,/usr/share/sonic/device/x86_64-dell_s6000_s1220-r0/Force10-S6000/qos.json.j2
```

The benefit is consistent usage for single pair and batch mode. The implementation gaps:
1. support multiple input files in -t


---
In reply to: [458425757](https://github.com/Azure/sonic-buildimage/pull/4941#discussion_r458425757) [](ancestors = 458425757)